ios/CodePush/CodePush.m

```diff
@@ -455,9 +455,12 @@ - (void)rollbackPackage
 {
     NSError *error;
     NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
-
-    // Write the current package's metadata to the "failed list"
-    [self saveFailedUpdate:failedPackage];
+    if (error) {
```

Comment:
This is known as a "fragile" error check. Methods that return a value and accept an error should *always* check the return value from the method first. 

E.g. 

```
NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
if (!failedPackage) {
    // handle error here.
}
```